ts/packages/anchor/src/coder/borsh/instruction.ts

```diff
@@ -303,11 +304,15 @@ class InstructionFormatter {
           const variants = typeDef.type.variants;
           const variant = Object.keys(data)[0];
           const enumType = data[variant];
+          const camelCaseVariant = camelCase(variant);
+          const enumVariant = variants.find((v) => camelCase(v.name) === camelCaseVariant);
+          const relevantFields = enumVariant?.fields as IdlField[] | undefined;
```

Comment:
Guess this assertion is correct, given the comment at the start of this if (`// Struct enum.`)?

Reply:
I think the logic to decide enum type

https://github.com/coral-xyz/anchor/blob/7cbdff657d3f71d9851e316fd4e77e59fc3ffd94/ts/packages/anchor/src/coder/borsh/instruction.ts#L302

 is incorrect because all variants have `name` field. I've taken a note for this and will fix it.